pkg/controller/storagecluster/reconcile.go

```diff
@@ -250,8 +242,8 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 	r.conditions = nil
 	// Start with empty r.phase
 	r.phase = ""
-
-	for _, f := range []func(*ocsv1.StorageCluster, logr.Logger) error{
+	// list of default enusure functions
```

Comment:
Instead of mixing and complicating things, can we separate concerns? This looks like a code maintenance nightmare to me.

Let's have `ensureConvergedCluster` and `ensureExternalCluster` functions that will just do what it is supposed to do? We can use if/else to diverge the flow wherever needed.

@jarrpa WDYT?

Reply:
@umangachapagain , may be I am not seeing the big picture here, but until you are doing different actions in the `for` loop for each type of storage clusters (external or converged), you don't have to make separate functions (as the actions taken on each iterations are same for any type storagecluster).

Reply:
At the moment I don't want separate reconciliation functions between converged and external clusters. I think the overhead of maintenance is worth it to make sure that we don't diverge greatly in execution between the two and ensure that we don't break one when modifying the other.